### ISSUE TITLE
RFO: what is our recommendation for assigning version numbers to genomes, annotations, etc.??

### ISSUE THREAD

**Issue by adf-ncgr** [LegumeFederation/datastore#21]
In LIS, we have sometimes taken the point of view that it is good to preserve the version numbers originally associated with datasets in our datastore naming, for example:
Trifolium_pratense/MilvusB.gnm2.gNmT
Vigna_radiata/VC1973A.gnm6.3nL8
Vigna_angularis/Gyeongwon.gnm3.3Nz5
(the point being that none of these start at gnm1 because the first version we had access to was
the number you see)

On the other hand, since the genome browser version for Pigeonpea has always been given as
"Cc1.0" I have been assuming we would release this in the datastore as:
cajca.ICPL87119.gnm1 and cajca.ICPL87119.gnm1.ann1 (since we never seem to use ann0)

On the other hand, looking at the version of the files posted here: http://gigadb.org/dataset/100028 
it seems they are V5.0 although this designation appears nowhere that I can see in the files hosted here:
http://www.ceg.icrisat.org/gt-bt/iipg/genomedata.zip

Based on cases such as these, I am inclined to throw up my hands and decide that in general we
need not care about trying to match external versions (except as a note in a README). There will likely be exceptions (I'd hate to have to tell Chris and Nevin that we are calling Mt4.0 medtr.A17.gnm1.ann1). Is there anything more clear policy that we can develop regarding this issue? 

NB: the actual RFO is that I propose to leave cajca.ICPL87119.gnm1.ann1 as is rather than go back and make it cajca.ICPL87119.gnm5.ann1

**Comment by cann0010** [comment]
I am inclined to try to stick with versioning from the sequencing group - e.g. Mt4.0 -> Medicago_truncatula/A17.gnm4 or Trifolium_pratense/MilvusB.gnm2
... but I acknowledge that we haven't always been consistent, so there will probably be some cases that are grandfathered as exceptions.

**Comment by adf-ncgr** [comment]
thanks- does that suggest we would also call the annotations:
Medicago_truncatula/A17.gnm4.ann0
?
believe it or not, I'm not trying to be pedantic here!

**Comment by vivekkrish** [comment]
Hi @adf-ncgr, @cann0010 -

I believe that since the annotation versions are Mt4.0v1, Mt4.0v2, etc.; they would correspondingly be called gnm4.ann1, gnm4.ann2, and so on.

**Comment by adf-ncgr** [comment]
thanks @vivekkrish -
OK, so maybe that was a bad example of the point I was trying to get at (thanks to the benevolent and wise keepers of medicagogenome for versioning things relatively unambiguously).
And please don't get me wrong, I very much like the idea of being consistent with what the sequencing group does, and have tried to follow this model in the past. However, I do find that it leads to difficulties of interpretation of what exactly should be done in specific cases, e.g. in the case of pigeonpea that I started with, which is just given as v5.0  (and not even consistently as that in various locations). Maybe it is not worth worrying about, but I still do.

I could live with the rule that if an annotation is bundled with a genome assembly and no version distinction is made, then we just call it ann1 (though I still don't want to backtrack and call pigeonpea: cajca.ICPL87119.gnm5.ann1). 

But we would still need to determine whether version designations with "." in them are taken as is, e.g. would we have used medtr.A17.gnm3.5.ann1 ? 

and it would be interesting to see if we were all to peruse the set of versions given by NCBI or phytozome whether we would all interpret them the same way, e.g. one of my favorite examples:
https://phytozome.jgi.doe.gov/pz/portal.html#!info?alias=Org_Czofingiensis_er
"This release of a chromosome-level genome assembly (ChrZofV5) and associated protein-coding gene annotations (v5.2.3.2) supports C. zofingiensis as an emerging model system for oleaginous green algae."
gnm5.ann2.3.2 ? 
OK, so it's not a legume. Neither is 
https://phytozome.jgi.doe.gov/pz/portal.html#!info?alias=Org_Sbicolor
"The present v3.1.1 release, comprising the v3.0.1 assembly and v3.1.1 gene set, is a modern annotation using resources used in the original v1.0 release (Sbi1 assembly and Sbi1.4 gene set)"
gnm3.0.1.ann0.1.0? that doesn't seem right but I don't know what would.

https://www.ncbi.nlm.nih.gov/genome/annotation_euk/all/ has its fair share of puzzling cases, too.

I would also note that the general principle of using the submitting group's versioning scheme doesn't have a clear application to things like maps, expression datasets and diversity datasets,
e.g. 
Phaseolus_vulgaris/mixed.map1.blair.7PMp
Phaseolus_vulgaris/G19833.gnm1.div1.BHxl
Medicago_sativa/medsa.mixed.tcp1.var1.8nLW/

maybe that's OK, but I just want to make sure we can explain how we recommend doing things to federation members who might step up to help populate the "data commons".

thanks!

**Comment by adf-ncgr** [comment]
@jd-campbell 
I thought the review of this thread yielded nothing to suggest anyone objected to my earlier comment:

> NB: the actual RFO is that I propose to leave cajca.ICPL87119.gnm1.ann1 as is rather than go back and make it cajca.ICPL87119.gnm5.ann1

The reason I am reluctant to do so is because we just made all the chado naming consistent with the former, including loading of expression data. Changing now after the RFO has been dormant would be a major PITA, IMHO!

I think it would not be the first instance where we are somewhat out of sync with original naming, but I could be wrong. And the browser is Cc1.0 so if I was led astray I blame whoever did that! :)

**Comment by jd-campbell** [comment]
Ok, I must of misunderstood. I thought the RFO had been resolved based on @can0010 comments 'I am inclined to try to stick with versioning from the sequencing group'.

I will keep the genome at gnm1 until further notice.

=========================
Jacqueline D. Campbell (Farrell), PhD
Iowa State University
jdjax@iastate.edu
##Sent from my iPhone
##Please excuse any spelling or grammar errors.
________________________________

**Comment by adf-ncgr** [comment]
yes, I was relying on the latter half of the same @cann0010 comment:

> ... but I acknowledge that we haven't always been consistent, so there will probably be some cases that are grandfathered as exceptions.

as well as the fact that V5.0 doesn't appear at all in what I have considered to be the original version of the data at:
http://www.ceg.icrisat.org/gt-bt/iipg/genomedata.zip

(and the http://gigadb.org/dataset/100028 version is less comprehensive in some sense; for example, it doesn't seem to have the annotation files that separate out the TEs into a separate component)

thanks for relenting! you are hereby granted one "grandfather this inconsistency for free" card :)

**Comment by jd-campbell** [comment]
@adf-ncgr @cann0010 

Can this issue be closed? 

https://legumeinfo.org/data/public/Cajanus_cajan/
